File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/common/SharedPreferencesAuthTokenProvider.java
package com.zancheema.android.pantry.common;

import android.content.Context;
import android.content.SharedPreferences;

public class SharedPreferencesAuthTokenProvider implements AuthTokenProvider {
    private static final String KEY_ACCESS_TOKEN = "access-token";

    private final SharedPreferences sharedPreferences;

    public SharedPreferencesAuthTokenProvider(Context context, String filename) {
        sharedPreferences = context.getSharedPreferences(filename, Context.MODE_PRIVATE);
    }

    @Override
    public void saveAccessToken(String accessToken) {
        sharedPreferences
                .edit()
                .putString(KEY_ACCESS_TOKEN, accessToken)
                .apply();
    }

    @Override
    public String getAccessToken() {
        return sharedPreferences.getString(KEY_ACCESS_TOKEN, "");
    }
}

File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/common/SharedPreferencesAuthTokenProvider.kt
package com.zancheema.android.pantry.common

import android.content.Context

private const val KEY_ACCESS_TOKEN = "access-token"

class SharedPreferencesAuthTokenProvider(context: Context, filename: String) : AuthTokenProvider {

    private val sharedPreferences = context.getSharedPreferences(filename, Context.MODE_PRIVATE)

    override fun saveAccessToken(accessToken: String) {
        sharedPreferences
            .edit()
            .putString(KEY_ACCESS_TOKEN, accessToken)
            .apply()
    }

    override fun getAccessToken(): String {
        return sharedPreferences.getString(KEY_ACCESS_TOKEN, "") ?: ""
    }

    override fun deleteAccessToken() {
        sharedPreferences.edit()
            .remove(KEY_ACCESS_TOKEN)
            .apply()
    }
}